### PR TITLE
Update Visa calculator for Saudi Arabia and Bahrain

### PR DESCRIPTION
**Pull request by 1pretz1**
Trello:
https://trello.com/c/WeCrtlRX/1222-check-if-you-need-a-uk-visa-visa-waiver-changes

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/flows/check_uk_visa_flow/outcomes/outcome_medical_y.erb [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+    You’ll need a visa to visit the UK
+  <% end %>
 <% end %>
 
 <% govspeak_for :body do %>
-  You should apply for a [Standard Visitor visa](/standard-visitor).
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    You must either apply for:
+
+    + an [EVW](/get-electronic-visa-waiver) - you can apply from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_medical_y.erb:13]: Change [EVW] to [electronic visa waiver (EVW)] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_school_y.erb [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
 
 <% govspeak_for :body do %>
-  You can apply for a [Parent of a Child Student visa](/parent-of-a-child-at-school-visa) if your child is under 12 and at an independent day school.
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    To visit them for up to 6 months you must either apply for:
 
-  <% if calculator.passport_country_is_estonia? || calculator.passport_country_is_latvia? %>
-    You must apply for your visa from the country you’re currently living in.
-  <% end %>
+    + an [EVW](/get-electronic-visa-waiver) - you can apply from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_school_y.erb:13]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_standard_visitor_visa.erb [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+    You’ll need a visa to come to the UK
+  <% end %>
 <% end %>
 
 <% govspeak_for :body do %>
-  Apply for a [Standard Visitor visa](/standard-visitor).
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    You must either apply for:
+
+    + an [EVW](/get-electronic-visa-waiver) - you can apply from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_standard_visitor_visa.erb:13]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_study_m.erb [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
 <% end %>
 
 <% govspeak_for :body do %>
   <%= render partial: 'stateless_or_refugee', locals: {calculator: calculator} %>
 
-  Apply for a [Standard Visitor visa](/standard-visitor) if you're studying for 6 months or less.
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    You must either apply for:
+
+    + an [EVW](/get-electronic-visa-waiver) - you can apply from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_study_m.erb:15]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_tourism_visa_partner.erb [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+  <% end %>
 <% end %>
 
 <% govspeak_for :body do %>
-  In most cases, you must apply for a [Standard Visitor visa](/standard-visitor).
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    In most cases you must either apply for:
 
-  ##When you might not need a visa
+    + an [EVW](/get-electronic-visa-waiver) - you can apply from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_tourism_visa_partner.erb:13]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
     You’ll need a visa to pass through the UK in transit
   <% end %>
 <% end %>
 
 <% govspeak_for :body do %>
-  <% if calculator.passport_country_in_electronic_visa_waiver_list? %>
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    You must either apply for:
+
+    an [EVW](/get-electronic-visa-waiver) - you can apply for this from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb:13]: Change:

an [EVW] 

to:

+ an [electronic visa waiver] 

Reason:

To make the link text more accessible. Also bullet markdown is missing.

File: app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
+    a [Visitor in Transit visa](/transit-visa) if you arrive on a flight and will pass through immigration control before you leave the UK
+  <% elsif calculator.passport_country_in_electronic_visa_waiver_list? %>
     You must either apply for:
 
     - an [EVW](/get-electronic-visa-waiver)
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb:18]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_transit_to_the_republic_of_ireland.erb [hunk 8]
```diff
@@ -11,11 +13,17 @@
...
+    You must meet the [Standard Visitor eligibility requirements](/standard-visitor).
 
     [Check what documents you need to bring](/government/publications/visitor-visa-guide-to-supporting-documents) to show officers at the UK border.
 
     If your passport does not have a personal ID number on the bio data page you’ll usually need to apply for a [Standard Visitor visa](/standard-visitor).
+  <% elsif calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+
+    You must either apply for:
+
+    + an [EVW](/get-electronic-visa-waiver) - you can apply for this from 1 June 2022
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_transit_to_the_republic_of_ireland.erb:25]: Change [EVW] to [electronic visa waiver] to make the link text more accessible.

File: app/flows/check_uk_visa_flow/outcomes/outcome_work_m.erb [hunk 9]
```diff
@@ -1,20 +1,40 @@
...
-  You can apply:
+    You can apply for:
+    + a [Standard Visitor visa](/standard-visitor) - if you’re coming to the UK for business activities like conferences, meetings, training, academic research or a sabbatical
+    + a [permitted paid engagement](/permitted-paid-engagement-visitor) visa if you’ve been invited as an expert in your profession - you can only stay for up to 1 month
 
-  + for a [Standard Visitor visa](/standard-visitor) - for example if you're coming to the UK for conferences, meetings, training, academic research or a sabbatical
-  + as a visitor doing a '[permitted paid engagement](/permitted-paid-engagement-visa)' (you must have been invited to the UK because of your expertise) - you can only stay for up to 1 month
+    ^
+    You will not need a visa if you hold a valid [electronic visa waiver (EVW)](/get-electronic-visa-waiver). You can apply for an EVW from 1 June 2022.
+    ^
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_work_m.erb:29]: Change lines 27-29 from:

^
You will not need a visa if you hold a valid [electronic visa waiver (EVW)](https://github.com/get-electronic-visa-waiver). You can apply for an EVW from 1 June 2022.
^

to:

^You will not need a visa if you hold a valid [electronic visa waiver (EVW)](https://github.com/get-electronic-visa-waiver). You can apply for an EVW from 1 June 2022.^

Reason:

Markdown is incorrect - there should be no line break after the first ^ and before the second ^.

File: app/flows/check_uk_visa_flow/outcomes/outcome_marriage_visa_nat_direct_airside_transit_visa.erb [hunk 10]
```diff
@@ -9,12 +9,24 @@
...
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    If you want to live in the UK permanently, you must apply for a [family visa](/uk-family-visa) if your partner is any of the following:
+
+    + a British citizen
+    + settled in the UK
+    + someone with humanitarian protection, for example if they have refugee status
+
+    ^
+    If you want to convert a civil partnership into a marriage, you’ll need to apply for an [electronic visa waiver (EVW)](/get-electronic-visa-waiver) or a [Standard Visitor visa](/standard-visitor).
+    ^
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_marriage_visa_nat_direct_airside_transit_visa.erb:21]: Change lines 19-21 from:

^
If you want to convert a civil partnership into a marriage, you’ll need to apply for an [electronic visa waiver (EVW)](/get-electronic-visa-waiver) or a [Standard Visitor visa](/standard-visitor).
^

to:

^If you want to convert a civil partnership into a marriage, you’ll need to apply for an [electronic visa waiver (EVW)](/get-electronic-visa-waiver) or a [Standard Visitor visa](/standard-visitor).^

Reason:

Markdown is incorrect - there should be no line break after the first ^ and before the second ^.

File: app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb [hunk 11]
```diff
@@ -1,13 +1,18 @@
...
   <% end %>
 <% end %>
 
 <% govspeak_for :body do %>
-  <% if calculator.passport_country_in_electronic_visa_waiver_list? %>
+  <% if calculator.passport_country_is_saudi_arabia_or_bahrain? %>
+    You must either apply for:
+
+    an [EVW](/get-electronic-visa-waiver) - you can apply for this from 1 June 2022
+    a [Visitor in Transit visa](/transit-visa) if you arrive on a flight and will pass through immigration control before you leave the UK
```
Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb:14]: Change:

a [Visitor in Transit visa](/transit-visa) if you arrive on a flight and will pass through immigration control before you leave the UK

to:

+ a [Visitor in Transit visa](/transit-visa) 

Reason:

Wording is incorrect. Also bullet markdown is missing.

Comment (mattclear) [inline app/flows/check_uk_visa_flow/outcomes/outcome_transit_leaving_airport.erb:14]: Change:

a [Visitor in Transit visa](/transit-visa) if you arrive on a flight and will pass through immigration control before you leave the UK

to:

+ a [Visitor in Transit visa](/transit-visa) 

Reason:

Wording is incorrect. Also bullet markdown is missing.

**Review by mattclear — CHANGES_REQUESTED** [review]
Hi, just a few minor comments on wording/markdown.

**Review by jon-kirwan — APPROVED** [review]
I've been through various scenarios to get to each of the updated outcome pages and all changes looks good to me 🚀

The only thing I did notice was related to the outcome page for `/flows/check_uk_visa_flow/outcomes/outcome_visit_waiver.erb/` - shown when tested with _Kuwait_. This doesn't seem to be included in the list of pages to be updated but the content is:

> You must either apply for:

  - an [EVW](/get-electronic-visa-waiver)
  - a [Standard Visitor visa](/standard-visitor)

and to be consistent with the other pages maybe it should be updated to:

> You must either apply for:

  - an [electronic visa waiver](/get-electronic-visa-waiver)
  - [a Standard Visitor visa](/standard-visitor) for courses up to 6 months

**Comment by mattclear** [comment]
@jon-kirwan Thanks for flagging that. We should definitely change the link text in the first bullet from 'EVW' to 'electronic visa waiver'.

'for courses up to 6 months' being included on that outcome in the second bullet was something we noticed earlier and queried with the department. Apparently it is correct, so should be left as it is.

**Comment by edwardkerry** [comment]
@mattclear I have made that last change as well to `/flows/check_uk_visa_flow/outcomes/outcome_visit_waiver.erb/`. Are you able to mark your requested changes as complete and we will then merge this in? (the merge button is unavailable until the changes are approved!)

**Review by mattclear — APPROVED** [review]

**Pull request merged**